Tasks/NotationV0/task.json

```diff
@@ -0,0 +1,251 @@
+      },
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "akvPluginVersion",
+      "type": "string",
+      "label": "Plugin Version",
+      "defaultValue": "1.0.1",
+      "required": true,
+      "helpMarkDown": "The version of the Azure Key Vault plugin to be installed. please visit the [release page](https://github.com/Azure/notation-azure-kv/releases) for the available versions.",
+      "visibleRule": "plugin = azureKeyVault",
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "azurekvServiceConection",
+      "type": "connectedService:AzureRM",
+      "label": "Azure Key Vault service connection",
+      "defaultValue": "",
+      "required": false,
+      "helpMarkDown": "Select the Azure subscription for the key vault if prefer to use service connection for authentication.",
+      "visibleRule": "plugin = azureKeyVault",
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "keyid",
+      "type": "string",
+      "label": "Key ID",
+      "required": true,
+      "defaultValue": "",
+      "helpMarkDown": "The Key ID is the key or certificate identifier for Azure Key Vault.",
+      "visibleRule": "plugin = azureKeyVault",
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "caCertBundle",
+      "type": "string",
+      "label": "Certificate Bundle File Path",
+      "required": false,
+      "defaultValue": "",
+      "visibleRule": "plugin = azureKeyVault",
+      "helpMarkDown": "A file with root and all intermediate certificates, starting from the root certificate, following the order in the certificate chain.",
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "selfSigned",
+      "type": "boolean",
+      "label": "Self-signed Certificate",
+      "required": false,
+      "defaultValue": false,
+      "visibleRule": "plugin = azureKeyVault",
+      "groupName": "pluginConfig"
+    },
+    {
+      "name": "trustPolicy",
+      "type": "string",
+      "label": "Trust Policy File Path",
+      "required": true,
+      "defaultValue": "",
+      "visibleRule": "command = verify",
+      "helpMarkDown": "The path to the [trust policy](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/trust-store-trust-policy.md#trust-policy) file relative to the repository. Example: ./path/to/trust-policy.json",
+      "groupName": "commandConfig"
+    },
+    {
+      "name": "trustStore",
+      "type": "string",
+      "label": "Trust Store Folder Path",
+      "required": true,
+      "defaultValue": "",
+      "visibleRule": "command = verify",
+      "helpMarkDown": "The path to the directory containing the [trust store](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/trust-store-trust-policy.md#trust-store) relative to the repository. Example: ./path/to/truststore/",
+      "groupName": "commandConfig"
+    }
+  ],
+  "execution": {
+    "Node16": {
+      "target": "src/index.js",
+      "argumentFormat": ""
+    },
+    "Node20": {
```

Comment:
it should be Node20_1

Reply:
Hi @kirill-ivlev , I'm not sure how the agent uses this field. I didn't find any other task with Node20_1 in this repo. I also tried updating the dependency to Node 20 and TypeScript 5.1.6. However, it seems like the build script still has some issues. The behavior is different on Linux and Windows, so I'm continuing to use Node 16 in this PR.

Also, I found this comment, which suggests that we might need to update the build script to support a native Node 20 task. I'm not entirely sure though.
https://github.com/microsoft/azure-pipelines-tasks/blob/1d6dbad5a08219769465bebb44ed2f53bc91b582/make.js#L240-L241
I

Reply:
Updated to use `Node20_1`

Reply:
Hi @JeyJeyGao, the node20_1 handler is needed because the node20 was renamed in [the agent's PR](https://github.com/microsoft/azure-pipelines-agent/pull/4518). 

Node 16 handler is needed for backward compatibility because we introduced node20 runner not so far ago(in version [3.225](https://github.com/microsoft/azure-pipelines-agent/releases/tag/v3.225.0), and if the customers have an old agent which doesn't have node20, the task will be run under node16.

The line you're pointing to in the make.js file is for tasks which were migrated using the build config generator, such tasks have [_buildConfig](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/ANTV1/_buildConfigs/Node20) directory inside and are presented in _generated folder with [Node20_1 handler](https://github.com/microsoft/azure-pipelines-tasks/blob/master/_generated/ANTV1_Node20/task.json#L261)

Also, since you don't use the generator (because the task is initially on node20 and won't be upgraded using codegen), I've updated typescript and @types/node packages

Reply:
Thank you @DmitriiBobreshev !